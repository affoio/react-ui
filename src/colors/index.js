const colors = {
  indigoBlue: {
    800: '#4718b4',
    700: '#623abf',
    600: '#8769ce',
    500: '#ac97dd',
    400: '#d1c5ec',
    200: '#ece7f7',
    300: '#e4ddf4',
    100: '#f6f3fb',
  },
  marigold: {
    800: '#ffca00',
    700: '#ffd226',
    600: '#ffdc59',
    500: '#ffe78c',
    400: '#fff2bf',
    300: '#fff7d9',
    200: '#fff9e5',
    100: '#fffcf2',
  },
  cerise: {
    800: '#e2086e',
    700: '#e62d84',
    600: '#ec5ea1',
    500: '#f290be',
    400: '#f8c1db',
    300: '#fbdaea',
    200: '#fce6f0',
    100: '#fdf2f8',
  },
  black: {
    800: '#000000',
    700: '#262626',
    600: '#595959',
    500: '#8c8c8c',
    400: '#bfbfbf',
    300: '#d9d9d9',
    200: '#e5e5e5',
    100: '#f2f2f2',
  },
  seaweed: {
    800: '#13c64f',
    700: '#36ce69',
    600: '#65da8c',
    500: '#95e5b0',
    400: '#c4f1d3',
    300: '#dcf7e5',
    200: '#e7f9ed',
    100: '#f3fcf6',
  },
  white: '#ffffff',
};

export default colors;
