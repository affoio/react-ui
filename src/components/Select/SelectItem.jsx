import React from 'react';
import { bool } from 'prop-types';
import styled from 'styled-components';
import { addStyleIfPropTruly } from '../../helpers/styledComponents';
import colors from '../../colors';
import { getMultipliedIndent } from '../../theme/selectors';

const SelectItemStyled = styled.div`
  height: ${({ theme }) => getMultipliedIndent(theme, 12)};
  box-sizing: border-box;
  width: 100%;
  display: flex;
  align-items: center;
  padding: 0 ${({ theme }) => getMultipliedIndent(theme, 4)};
  cursor: pointer;

  &:hover {
    background-color: ${colors.indigoBlue[100]};
  }

  ${addStyleIfPropTruly('focused', `background-color: ${colors.indigoBlue[100]};`)}
  ${addStyleIfPropTruly('checked', `background-color: ${colors.indigoBlue[200]};`)}
  ${addStyleIfPropTruly('inline', 'white-space: nowrap;')}
`;

const propTypes = {
  checked: bool,
};

const SelectItem = React.forwardRef((props, ref) => <SelectItemStyled {...props} ref={ref} />);

SelectItem.propTypes = propTypes;

export default SelectItem;
