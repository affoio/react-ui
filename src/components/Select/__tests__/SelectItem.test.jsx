import React from 'react';
import renderer from 'react-test-renderer';
import SelectItem from '../SelectItem';
import colors from '../../../colors';

describe('<SelectItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SelectItem />);

    expect(tree).toMatchSnapshot();
  });

  it('changes color when checked', () => {
    const tree = renderer.create(<SelectItem checked />).toJSON();

    expect(tree).toHaveStyleRule(`background-color: ${colors.indigoBlue[100]}`);
  });
});
