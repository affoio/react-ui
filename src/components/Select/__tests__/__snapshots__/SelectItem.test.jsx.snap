// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectItem renders correctly 1`] = `
.c0 {
  height: 48px;
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: #f6f3fb;
}

<div
  className="c0"
/>
`;
